### PR TITLE
59 > My Programmes

### LINKED ISSUE #59: I can view a list of home programmes
__Wireframe link__

https://www.figma.com/file/CcYmhfnXreAPxlfyEmGsAH/Chiltern-Music-Therapy?node-id=892%3A157

---

### Acceptance Criteria:

_REMEMBER THAT WHOEVER WORKS ON THIS ISSUE MUST TICK OFF ALL THE POINTS IN THIS LIST UNLESS THERE IS CLEAR AGREEMENT IN THE COMMENTS TO SAY OTHERWISE. **DO NOT REVIEW A PR INVOLVING THIS ISSUE UNLESS THIS HAS BEEN DONE**_ 

- [x] Create page in line with wireframes
- [x] Api call and db query to get ALL programmes linked with the client
- [ ] For now there is no lazy loading or pagination - it will get all programmes and list them on the page

### PR DESCRIPTION
**Pull request by thejoefriel**
#### Issues this PR relates to
Closes #59 

=======

_NOTE: REMEMBER NOT TO REVIEW UNLESS ALL ACCEPTANCE CRITERIA HAS BEEN TICKED OFF_

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)

- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/pages/Client/AllProgrammes/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      }
+    };
+
+    getProgrammes();
+  }, []);
+
+  const programmesToView = programmes.length > 0;
+
+  return (
+    <>
```
Comment (fadeomar) [inline client/src/pages/Client/AllProgrammes/index.js:31]: the width of cards and the paragraph doesn't match the wireframe, something I've faced on my PR too 
image
my solution was but now we should make  common solution for this 
```suggestion
    <div style={{maxWidth: 1065}}>
```

File: client/src/App.js [hunk 2]
```diff
@@ -44,6 +44,16 @@ function App() {
...
                 Component={Pages.THERAPIST.CreateProgram}
               />
+              <Route
+                exact
+                path={navRoutes.CLIENT.PROGRAMMES}
+                Component={Pages.CLIENT.AllProgrammes}
+              />
+              <Route
+                exact
+                path={navRoutes.CLIENT.PROGRAMMES}
```
Comment (fadeomar) [inline client/src/App.js:54]: this should be 
```suggestion
                path={navRoutes.THERAPIST.CREATE_PROGRAM}
```

**Review by fadeomar — CHANGES_REQUESTED** [review]

File: client/src/components/Cards/Link/index.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ const Link = ({
   switch (variant) {
     case 'graphic1':
       return (
-        <ReactLink to={to}>
+        <ReactLink to={to} style={{ width: '100%' }}>
```
[the remark this answers is not in the document]
Comment (thejoefriel) [reply client/src/components/Cards/Link/index.js:21]: done

File: client/src/pages/Client/AllProgrammes/index.js (hunk 1 above)
Comment (thejoefriel) [reply client/src/pages/Client/AllProgrammes/index.js:31]: It's fine that the cards and paragraph change width slightly (it doesn't need to always be exactly 300px as that's not always ideal for a responsive design). I agree that there should be a max-width on the content in the layout though for extra wide screens (that was already meant to be in there from Figma as I remember discussing this with Jem but maybe was missed )

File: client/src/App.js (hunk 2 above)
Comment (thejoefriel) [reply client/src/App.js:54]: removed duplicate

**Review by fadeomar — APPROVED** [review]

**Pull request merged**